ModelLibrary/Process/CreateContractInvoice.cs

```diff
@@ -328,5 +343,30 @@ private void GenerateInvoice(VAdvantage.Model.X_C_ContractSchedule contSchedule)
                 //_count++;
             }
         }
+        /// <summary>
+        /// Created By: Rakesh Kumar
+        /// Created Date: 01/Apr/2021
+        /// Purpose: Get DocType
+        /// </summary>
+        private void SetDocType()
+        {
+            _C_DocType_ID = 0;
+            // When ContractType is Accounts Receivable
+            if (cont.GetContractType() == X_C_Contract.CONTRACTTYPE_AccountsReceivable)
+            {
+                sql.Append(MRole.GetDefault(GetCtx()).AddAccessSQL("SELECT MIN(C_DOCTYPE_ID) FROM C_DocType WHERE DOCBASETYPE='ARI' AND ISACTIVE ='Y'", "C_DocType", true, true));
```

Comment:
Use function from MDocType to get document type based on Document Base Type

Reply:
Changes Done